components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -91,13 +92,12 @@ public class OAuth2Service extends AbstractAdmin {
      * Process the authorization request and issue an authorization code or access token depending
      * on the Response Type available in the request.
      *
-     * @param oAuth2AuthorizeReqDTO <code>OAuth2AuthorizeReqDTO</code> containing information about the authorization
-     *                              request.
+     * @param authzReqMsgCtx authzReqMsgCtx.
      * @return <code>OAuth2AuthorizeRespDTO</code> instance containing the access token/authorization code
      * or an error code.
      */
-    public OAuth2AuthorizeRespDTO authorize(OAuth2AuthorizeReqDTO oAuth2AuthorizeReqDTO) {
-
+    public OAuth2AuthorizeRespDTO authorize(OAuthAuthzReqMessageContext authzReqMsgCtx) {
```

Comment:
Changing the public method without version bump is not a good practice.

Can we deprecate the old method and introduce this as a new method?